submit.go

```diff
@@ -165,14 +165,26 @@ type submitResponse struct {
 	ReportURL string `json:"report_url,omitempty"`
 }
 
+type submitErrorResponse struct {
+	Error string `json:"error"`
+	ErrorCode string `json:"errcode"`
+}
+
+type submitPolicyErrorResponse struct {
+	Error string `json:"error"`
+	ErrorCode string `json:"errcode"`
+	PolicyUrl string `json:"policy_url,omitempty"`
+}
+
+
 func (s *submitServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	// if we attempt to return a response without reading the request body,
 	// apache gets upset and returns a 500. Let's try this.
 	defer req.Body.Close()
 	defer io.Copy(ioutil.Discard, req.Body)
 
 	if req.Method != "POST" && req.Method != "OPTIONS" {
-		respond(405, w)
+		writeError(w, 405, submitErrorResponse{"Method not allowed. Use POST.", "RS_METHOD_NOT_ALLOWED"})
```

Comment:
could you define some `const`s for the error codes, to collect them all together?

Reply:
done